os/drivers/wireless/realtek/wifi_util_interface/wifi_interactive_mode.c

```diff
@@ -772,91 +772,107 @@ static void print_scan_result(rtw_scan_result_t *record)
 
 #ifdef CONFIG_PLATFORM_TIZENRT
 #include <../../framework/src/wifi_manager/wifi_common.h>
-wifi_utils_scan_list_s *scan_list = NULL;
-wifi_utils_scan_list_s *_scan_list = NULL;
 
-extern int8_t wifi_scan_result_callback(wifi_utils_scan_list_s *scan_list);
+/* Scanning operation should be blocked mode */
+wifi_utils_scan_list_s *g_scan_list;
+int g_scan_num;
+static void _free_scanlist(void)
+{
+	while (g_scan_list) {
+		wifi_utils_scan_list_s *cur = g_scan_list;
+		g_scan_list = g_scan_list->next;
+		rtw_mfree(cur, sizeof(cur));
```

Comment:
Null Check!

Reply:
Thanks for the comment. While loop itself already checks the nullity of g_scan_list. Therefore, (wifi_utils_scan_list_s *) cur becomes never NULL here.

Reply:
Ok I missed it.

Reply:
Ok I missed it.